src/main/java/project/onlinebookstore/model/Book.java

```diff
@@ -7,11 +7,19 @@
 import jakarta.persistence.Id;
 import jakarta.persistence.Table;
 import java.math.BigDecimal;
-import lombok.Data;
+import lombok.Getter;
+import lombok.RequiredArgsConstructor;
+import lombok.Setter;
+import org.hibernate.annotations.SQLDelete;
+import org.hibernate.annotations.Where;
 
+@Setter
+@Getter
+@RequiredArgsConstructor
+@SQLDelete(sql = "UPDATE books SET is_deleted = true WHERE id =?")
+@Where(clause = "is_deleted = false")
```

Comment:
```suggestion
@SQLRestriction(value = "is_deleted = false")
```

`@Where` is deprecated